swagger/index.js

```diff
@@ -318,12 +328,24 @@ module.exports = yeoman.Base.extend({
     function generateRemoteMethods(self, cb) {
       var apis = self.apis;
       async.eachSeries(apis, function(api, done) {
-        var modelDef = api.modelDefinition;
-        if (!modelDef) {
-          return done();
-        }
-        self.log(chalk.green(g.f('Generating %s', modelDef.scriptPath)));
-        fs.writeFile(modelDef.scriptPath, api.code, done);
+        async.forEachOf(api.code, function (code, m, done) {
+          if (self.selectedModels[m] !== SELECTED_FOR_UPDATE &&
+            self.selectedModels[m] !== SELECTED_FOR_CREATE) {
+            return done();
+          }
+          var modelDef = api.modelDefinition;
+          for (var i = 0, n = self.modelDefs.length; i < n; i++) {
+            if (m === self.modelDefs[i].name) {
+              modelDef = self.modelDefs[i];
+              break;
+            }
```

Comment:
Oh, I see, you are defaulting to `api.modelDefinition` if no matching `self.modelDefs[i].name` is found.

How about the following then?

```js
const modelDef = self.modelDefs.find(d => d.name === m) ||
  api.modelDefinition;
```

My point here is that iterating arrays should be avoided in favour of `Array.prototype` functions - they make the code much easier to understand and prevent a whole range of errors.